pyproject.toml

```diff
@@ -19,3 +100,165 @@ skip = '.git,*.pdf,*.svg,versioneer.py,*.css,test_*'
 check-hidden = true
 ignore-regex = '^\s*"image/\S+": ".*|\b[Mm]anuel[. ][Hh]oltgrewe\b'
 ignore-words-list = 'testin'
+
+####################################################################
+# Pixi Configuration
+####################################################################
+
+[tool.pixi.project]
+channels = ["conda-forge", "bioconda"]
+platforms = ["linux-64", "osx-arm64", "win-64", "osx-64", "linux-aarch64"]
+
+[tool.pixi.pypi-dependencies]
+snakemake = { path = ".", editable = true }
+
+[tool.pixi.environments]
+dev = { features = [
+  # optional dependencies
+  "messaging",
+  "pep",
+  "reports",
+  # plugins
+  "plugin-interfaces",
+  "storage-plugins",
+  "executor-plugins",
+  # dev 
+  "test",
+  "docs",
+  "style",
+] }
+test = { features = [
+  "test",
+  "plugin-interfaces",
+  "storage-plugins",
+  "executor-plugins",
+] }
+docs = { features = ["docs", "plugin-interfaces"] }
+quality = { features = ["style"] }
+publish = { features = ["publish"] }
+
+#----------------------- Plugins -----------------------------
+
+# Plugin interfaces
+[tool.pixi.feature.plugin-interfaces.pypi-dependencies]
+snakemake-interface-executor-plugins = ">=9.3.2, <10"
+snakemake-interface-storage-plugins = ">=3.3.0, <4"
+
+# Storage plugins
+[tool.pixi.feature.storage-plugins.pypi-dependencies]
+snakemake-storage-plugin-http = "*"
+snakemake-storage-plugin-s3 = "*"
+snakemake-storage-plugin-fs = "*"
+
+# Executor plugins
+[tool.pixi.feature.executor-plugins.pypi-dependencies]
+snakemake-executor-plugin-cluster-generic = "*"
+
+#------------------------ Test -------------------------------  
+
+[tool.pixi.feature.test.dependencies]
+pytest = ">=8.3.5,<9"
+pytest-mock = ">=3.14.0,<4"
+pytest-cov = ">=6.0.0,<7"
+pytest-split = ">=0.10.0,<0.11"
+graphviz = ">=12.2.1,<13"
+oauth2client = ">=4.1.3,<5"
+numpy = ">=2.2.3,<3"
+glpk = ">=5.0,<6"
+pulp = ">=2.8.0,<3"
+boto3 = ">=1.37.9,<2"
+packaging = ">=24.2,<25"
+reretry = ">=0.11.8,<0.12"
+smart_open = ">=7.1.0,<8"
+pyyaml = ">=6.0.2,<7"
+responses = ">=0.25.6,<0.26"
+gitpython = ">=3.1.44,<4"
+pytools = ">=2024.1.6,<2026"
+pandoc = ">=3.6.3,<4"
+crc32c = ">=2.7.1,<3"
+nbformat = ">=5.10.4,<6"
+filelock = ">=3.17.0,<4"
+humanfriendly = ">=10.0,<11"
+pandas = ">=2.2.3,<3"
+pygments = ">=2.19.1,<3"
+wrapt = ">=1.17.2,<2"
+psutil = ">=7.0.0,<8"
+throttler = ">=1.2.2,<2"
+immutables = ">=0.21,<0.22"
+pysftp = ">=0.2.9,<0.3"
+yte = ">=1.5.7,<2"
+ftputil = ">=5.1.0,<6"
+requests = ">=2.32.3,<3"
+dpath = ">=2.2.0,<3"
+httpretty = ">=1.1.4,<2"
+appdirs = ">=1.4.4,<2"
+polars = ">=1.24.0,<2"
+jsonschema = ">=4.23.0,<5"
+requests-mock = ">=1.12.1,<2"
+docutils = ">=0.20.1,<0.22"
+tabulate = ">=0.9.0,<0.10"
+configargparse = ">=1.7,<2"
+
+[tool.pixi.feature.test.target.linux-64.dependencies]
+environment-modules = "*"
+xorg-libxrender = "*"
+xorg-libxext = "*"
+xorg-libxau = "*"
+xorg-libxdmcp = "*"
+xorg-libsm = "*"
+stress-ng = "*"
+nodejs = "*"
+apptainer = "*"
+squashfuse = "*"
+cwltool = "*"
+cwl-utils = "*"
```

Comment:
Some of these are already normal dependencies of Snakemake. Hence, they can be removed here. Example: configargparse.